### ISSUE TITLE
Feature request: automatically use `qemu` to emulate architectures

### ISSUE THREAD

**Issue by DilumAluthge** [staticfloat/Sandbox.jl#47]
The context here is that I am running the `test_rootfs.jl` script from the [`rootfs-images`](https://github.com/JuliaCI/rootfs-images) repository.

I'm on an `x86_64` Linux machine, and I am using the `UserNamespacesExecutor` executor (specifically, the `UnprivilegedUserNamespacesExecutor` executor).

I'm able to enter the `package_linux.x86_64` and `package_linux.i686` rootfs images and run commands inside them without any problems.

However, I'd also like to be able to enter rootfs images for other architectures, such as:
- `package_linux.aarch64`
- `package_linux.armv7l`
- `package_linux.powerpc64le`

Since I only have access to `x86_64` Linux machines, I'd like use QEMU (in user mode) to enter these rootfs images and run commands inside them.

Could we add some keyword arguments that I can pass to Sandbox.jl that will tell it to automatically add the appropriate `qemu` commands for me, so that I don't have to do it manually?

E.g. something like:
```julia
with_executor(UnprivilegedUserNamespacesExecutor; qemu = true, arch = "aarch64")
```

**Comment by DilumAluthge** [comment]
For example, this is what I currently get if I try to enter the `package_linux.aarch64` rootfs image on an `x86_64` Linux machine:

```
dilum@amdci6:~/Desktop/rootfs-images$ julia --project test_rootfs.jl https://github.com/JuliaCI/rootfs-images/releases/download/v3.3/package_linux.aarch64.tar.gz eb0b98a0207b88a748adb7e1d5318f17d37845a8
ERROR: Failed to run /bin/bash: 8 (Exec format error)
ERROR: LoadError: failed process: Process(setenv(`/home/dilum/.julia/artifacts/d71a9ea85301d090012dfa3bfcb16ea11bcdaf88/bin/sandbox --rootfs /home/dilum/.julia/artifacts/eb0b98a0207b88a748adb7e1d5318f17d37845a8 --cd / --uid 26175 --gid 26175 -- /bin/bash`,["PATH=/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/bin:/sbin", "USER=juliaci", "HOME=/home/juliaci"]), ProcessExited(1)) [1]

Stacktrace:
 [1] pipeline_error
   @ ./process.jl:525 [inlined]
 [2] run(::Base.CmdRedirect; wait::Bool)
   @ Base ./process.jl:440
 [3] run
   @ ./process.jl:438 [inlined]
 [4] run(exe::UnprivilegedUserNamespacesExecutor, config::SandboxConfig, user_cmd::Cmd; kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ Sandbox ~/.julia/packages/Sandbox/ylKFa/src/Sandbox.jl:120
 [5] run
   @ ~/.julia/packages/Sandbox/ylKFa/src/Sandbox.jl:115 [inlined]
 [6] (::var"#1#2")(exe::UnprivilegedUserNamespacesExecutor)
   @ Main ~/Desktop/rootfs-images/test_rootfs.jl:39
 [7] with_executor(f::var"#1#2", ::Type{UnprivilegedUserNamespacesExecutor}; kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ Sandbox ~/.julia/packages/Sandbox/ylKFa/src/Sandbox.jl:132
 [8] with_executor(f::var"#1#2", ::Type{UnprivilegedUserNamespacesExecutor}) (repeats 2 times)
   @ Sandbox ~/.julia/packages/Sandbox/ylKFa/src/Sandbox.jl:130
 [9] top-level scope
   @ ~/Desktop/rootfs-images/test_rootfs.jl:38
in expression starting at /home/dilum/Desktop/rootfs-images/test_rootfs.jl:38
```

**Comment by DilumAluthge** [comment]
Oh, also, we should document the steps that the user needs to take before using this feature. E.g. "install QEMU on the host system".

**Comment by staticfloat** [comment]
Fixed by #49
